src/client/ui/EditorContainer.js

```diff
@@ -428,7 +429,17 @@ class EditorContainer extends Component {
   onOpenScene = async uri => {
     if (this.props.editor.sceneInfo.uri === uri) return;
     if (!this.confirmSceneChange()) return;
+    const action = "openRootScene";
+    this._tryLoadSceneFromURI(uri, action, this.onOpenScene);
+  };
 
+  onExtendScene = async uri => {
+    if (!this.confirmSceneChange()) return;
+    const action = "extendScene";
+    this._tryLoadSceneFromURI(uri, action, this.onExtendScene);
```

Comment:
Would have been better to just pass in the function itself, instead of just the function name. Generally good to avoid dynamic, string-based lookups like this where possible. 
i.e:
```javascript
this._tryLoadSceneFromURI(uri, this.props.editor.extendScene.bind(this.props.editor), this.onExtendScene)
```

or just 

```javascript
this._tryLoadSceneFromURI(uri, this.props.editor.extendScene, this.onExtendScene)
```
if you convert `extendScene` to and arrow function first.